components/dom_crawler.rst

```diff
@@ -620,6 +620,22 @@ the whole form or specific field(s)::
     $form->disableValidation();
     $form['country']->select('Invalid value');
 
+Expanding URI
+~~~~~~~~~~~~~
+
+.. versionadded:: 5.1
+
+    The UriExpander helper was added in Symfony 5.1.
+
+The :class:`Symfony\\Component\\DomCrawler\\UriExpander` class provides method
+to expand a relative link::
```

Comment:
My biggest problem with this feature is the use of the "expand" word. First, I don't know what does exactly "expand" means (turning a relative URL into an absolute URL? turning anything into an absolute URL? etc.)

And ... if you look for "expand URI" ... you only find references to https://tools.ietf.org/html/rfc6570 which is related to "URI templates", where "expanding" means something very specific ... but unrelated to this feature.

Reply:
My English  skills was a bit too low on this one. 
To me, "expand something" means make it longer 
So expand an URL means make it longer... so absolute according to a base URL.

I can change the naming in Symfony. What do you suggest?

Reply:
I still don't understand precisely what does this feature achieve. So, before finding a new name, let's focus on describing what the feature does. Please 🙏 fill in the gaps here:

```
The UriExpander takes a __________ and transforms it into a ______________
```

Thanks!

Reply:
The UriExpander takes a **something** and transforms it into an **absolute URL according to a base URL**.

I said something because it can be one of:

* `a` 
* `a.html`
* `/a.html`
* `../a.html`
* `https://symfony.com/a.html`

When your browser render a link, the link (the href) can be anything. And the browser will  "expland" the thing to an absolute URL. In the HTML you can put literally anything. 

The goal here is to simulate what the browser do to the the absolute URL.

To get a full list of "thing" supported, you can look the test suite: https://github.com/symfony/symfony/pull/35415/files#diff-e06be322fcdda1bedb4ff51764aba2e6R27

Reply:
I was looking at classes like https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=netframework-4.8 and https://docs.oracle.com/javase/7/docs/api/java/net/URI.html ... and I think Java got a better name for this: "resolve". For example:

```
The UriResolver class takes an URI (relative, absolute, fragment, etc.)
and turns it into an absolute URI using another given base URI.
```

Reply:
I like resolves. I will do a PR

Reply:
Here we go https://github.com/symfony/symfony/pull/35667